parser/worker.go

```diff
@@ -336,7 +338,10 @@ func (w Worker) ExportTxs(txs []*types.Tx) error {
 		}
 
 		// call the tx handlers
-		go w.handleTx(tx)
+		err = w.handleTx(tx)
+		if err != nil {
+			return fmt.Errorf("error while handling transaction: %s", err)
+		}
```

Comment:
This is making multiple transactions handling synchronous again. What if instead we use the same `WaitGroup` that is used for messages, also to wait for this goroutine? This way we could have transaction and messages being handled concurrently, but still make sure everything completes before returning from `ExportTxs`.